cocoapods-core.gemspec

```diff
@@ -19,22 +19,17 @@ Gem::Specification.new do |s|
   s.files         =  Dir["lib/**/*.rb"] + %w{ README.md LICENSE }
   s.require_paths =  %w{ lib }
 
-  # 6.0 requires Ruby 2.5.0
-  s.add_runtime_dependency 'activesupport', '> 5.0', '< 6'
+  s.add_runtime_dependency 'activesupport', '~> 6.0'
```

Comment:
is there any reason we can't support both 5 and 6?

Reply:
In therory CocoaPods both support AS 5.x and 6.x, do you want both support it?

BTW, AS 5.3 may have protential compatibilty issue with Ruby 3.0+, because Rails 5.2 has report that can't running on Ruby 3.0

Reply:
I dont think Core actually uses activesupport?

```git grep activesupport
CHANGELOG.md:* Fixes crash when using plugins where activesupport 4 was not installed.  
Gemfile.lock:      activesupport (> 5.0, < 6)
Gemfile.lock:    activesupport (5.2.4.4)
cocoapods-core.gemspec:  s.add_runtime_dependency 'activesupport', '> 5.0', '< 6'
```

Reply:
Can we just remove this dependency?

Reply:
Let me try, I don't dig too much

Reply:
Core are using AS

![image](https://user-images.githubusercontent.com/1024162/104760216-024c4680-579c-11eb-9a27-bc48f366f0a8.png)

Reply:
ah, I was wrong in my search my bad

Reply:
@dnkoutso @segiddins  @jasl I recall we wanted to support latest major, but lower Ruby prevented this. Do you think it's better to have `'>= 5, < 7'`?

Reply:
AS 6 is designed for Ruby 2.5+, and the PR intend to set minimal Ruby to 2.6, so AS 6 is perfect suit,
what reason we need to keep compatibility with AS 5 ?

Reply:
Some folks might still be using cocoapods-core in rails 5 apps

Reply:
Got it, I'll loosen the version constraint

Reply:
Done, 
FYI: CI doesn't test against cocoapods-core with AS 5

Reply:
Just found an AS 6 compatibility issue on CocoaPods.
Good news is it's easy to fix

See https://github.com/CocoaPods/CocoaPods/pull/10364